anomalib/base/__init__.py

```diff
@@ -0,0 +1,19 @@
+"""Base classes for all anomaly components."""
+
+# Copyright (C) 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+from .anomaly_models import AnomalyModule, DynamicBufferModule
+
+__all__ = ["DynamicBufferModule", "AnomalyModule"]
```

Comment:
Not sure if we have any plans to add more base classes other than model, which is currently what we only have. 

I think this `base` would still be in located `anomalib.models.components`, while being imported via `from anomalib import AnomalyModule`.

Reply:
Does it make sense? I'm open to discussion here

Reply:
Looking at it now, I'm not sure if we should import the AnomalyModule from `anomalib` root level, while importing all other components from `anomalib.models.components`. This is from `padim/model.py`:
```
from anomalib import AnomalyModule
from anomalib.models.components import FeatureExtractor, MultiVariateGaussian
```
Which may look a bit out of line with how the repo is structured (AnomalyModule and FeatureExtractor are actually located at the same level in the folder hierarchy). I think it would be more consistent to import AnomalyModule from `anomalib.models`, since it is a base class of which all implementations also live in the `anomalib.models` directory.